go/vt/wrangler/testlib/reparent_external_test.go

```diff
@@ -300,7 +300,75 @@ func TestTabletExternallyReparentedFailedOldMaster(t *testing.T) {
 		t.Fatalf("GetTablet(%v) failed: %v", oldMaster.Tablet.Alias, err)
 	}
 	if tablet.Type != topodatapb.TabletType_REPLICA {
-		t.Fatalf("old master should be spare but is: %v", tablet.Type)
+		t.Fatalf("old master should be replica but is: %v", tablet.Type)
+	}
+}
+
+func TestTabletExternallyReparentedImpostorMaster(t *testing.T) {
+	tabletmanager.SetReparentFlags(time.Minute /* finalizeTimeout */)
+
+	ctx := context.Background()
+	ts := memorytopo.NewServer("cell1", "cell2")
+	wr := wrangler.New(logutil.NewConsoleLogger(), ts, tmclient.NewTabletManagerClient())
+
+	// Create an old master, a new master, and a bad slave.
+	badSlave := NewFakeTablet(t, wr, "cell1", 2, topodatapb.TabletType_MASTER, nil)
+	// do this after badSlave so that the shard record has the expected master
+	oldMaster := NewFakeTablet(t, wr, "cell1", 0, topodatapb.TabletType_MASTER, nil, ForceInitTablet())
+	newMaster := NewFakeTablet(t, wr, "cell1", 1, topodatapb.TabletType_REPLICA, nil)
+
+	// check the old master is really master
+	tablet, err := ts.GetTablet(ctx, oldMaster.Tablet.Alias)
+	if err != nil {
+		t.Fatalf("GetTablet(%v) failed: %v", oldMaster.Tablet.Alias, err)
+	}
+	if tablet.Type != topodatapb.TabletType_MASTER {
+		t.Fatalf("old master should be MASTER but is: %v", tablet.Type)
+	}
+
+	// On the elected master, we will respond to
+	// TabletActionSlaveWasPromoted.
+	newMaster.StartActionLoop(t, wr)
+	defer newMaster.StopActionLoop(t)
+
+	// On the old master, we will only respond to
+	// TabletActionSlaveWasRestarted.
+	oldMaster.StartActionLoop(t, wr)
+	defer oldMaster.StopActionLoop(t)
+
+	// On the bad slave, we will respond to
+	// TabletActionSlaveWasRestarted.
+	badSlave.StartActionLoop(t, wr)
```

Comment:
@enisoc I was trying to add one more test - where an impostor does not respond to RefreshState. I tried removing these two lines but that didn't have the desired effect. What is the right way to do that?

Reply:
Hm that should work to make it not respond. What effect were you looking for? All we do is log in that case, right?

Reply:
I was looking for a deadline exceeded error in the log, but didn't get one.

Reply:
Hm maybe it's related to this?

https://github.com/vitessio/vitess/blob/767e15767cee79a778923787695aada720d6b6e6/go/vt/wrangler/testlib/reparent_external_test.go#L255-L259

Reply:
Ah! copy-pasta error. Thanks.